drivers/winc1500/winc1500_netdev.c

```diff
@@ -0,0 +1,250 @@
+static void winc1500_isr(void *arg);
+
+/** netdev interface */
+static void _isr(netdev_t *dev);
+static int _init(netdev_t *dev);
+static int _send(netdev_t *netdev, const struct iovec *vector, unsigned count);
+static int _recv(netdev_t *netdev, void *buf, size_t len, void *info);
+static int _get(netdev_t *dev, netopt_t opt, void *value, size_t max_len);
+
+const netdev_driver_t netdev_driver_winc1500 = {
+    .init = _init,
+    .isr = _isr,
+    .send = _send,
+    .recv = _recv,
+    .get = _get,
+    .set = netdev_eth_set,
+};
+
+
+static void winc1500_isr(void *arg)
+{
+    winc1500_t *dev = (winc1500_t *) arg;
+
+    m2m_hif_isr();
+
+    /* call netdev hook */
+    dev->netdev.event_callback((netdev_t*) dev, NETDEV_EVENT_ISR);
+}
+
+
+static void _isr(netdev_t *netdev)
+{
+    winc1500_t *dev = (winc1500_t *) netdev;
+
+    _lock_bus(dev);
+    /* Handling events will be done in the callback */
+    m2m_wifi_handle_events(NULL);
+    _unlock_bus(dev);
+}
+
+
+static int _init(netdev_t *netdev)
+{
+    winc1500_t *dev = (winc1500_t *) netdev;
+
+    DEBUG("winc1500: starting initialization...\n");
+    /* Set SPI first */
+    dev->rx_size = -1;
+    spi_init(dev->params.spi);
+    _lock_bus(dev);
+
+    /* Initialize the WINC1500 BSP. */
+    dev->pid = thread_getpid();
+    nm_bsp_init();
+    nm_bsp_register_isr(winc1500_isr);
+
+    tstrWifiInitParam wifi_param;
+    memset((uint8_t *)&wifi_param, 0, sizeof(tstrWifiInitParam));
+
+    /* Register a callback. But replaced by the custom callback.
+     *  See process_event() in winc1500_callback.c */
+    wifi_param.pfAppWifiCb = NULL;
+    /* Setting Ethernet bypass mode. The ethernet buffer and callback will
+     *  be managed by netdev driver */
+    wifi_param.strEthInitParam.u8EthernetEnable = M2M_WIFI_MODE_ETHERNET;
+    wifi_param.strEthInitParam.au8ethRcvBuf = NULL;
+    wifi_param.strEthInitParam.u16ethRcvBufSize = 0;
+    wifi_param.strEthInitParam.pfAppEthCb = NULL;
+
+    if (M2M_SUCCESS != m2m_wifi_init(&wifi_param)) {
+        return WINC1500_ERR;
+    }
+
+    /* Register WINC1500's internal wifi callback instead of the driver's callback
+     *  for RIOT GNRC
+     */
+    hif_register_cb(M2M_REQ_GROUP_WIFI, _wifi_cb);
+
+    /* Initialize winc1500 struct */
+    mutex_init(&dev->mutex);
+    mbox_init(&dev->event_mbox, _mbox_msgs, WINC1500_EVENT_MBOX_LEN);
+
+    dev->state = 0 | WINC1500_STATE_INIT;
+    dev->state |= WINC1500_STATE_IDLE;
+
+    _unlock_bus(dev);
+
+#ifdef MODULE_NETSTATS_L2
+    memset(&netdev->stats, 0, sizeof(netstats_t));
+#endif
+    return 0;
+}
+
+
+static int _send(netdev_t *netdev, const struct iovec *vector, unsigned count) {
+    winc1500_t * dev = (winc1500_t *) netdev;
+    _lock_bus(dev);
+
+    /* Send packet */
+    size_t len = 0;
+    for (int i = 0; i < count; i++) {
+        m2m_wifi_send_ethernet_pkt(vector[i].iov_base, vector[i].iov_len);
+        len += vector[i].iov_len;
+        /* TODO: what if the module's buffer full? */
```

Comment:
Good question ;)

Reply:
I am working on this :) Please note that it is WIP for this moment.